src/go/rpk/pkg/adminapi/api_partition.go

```diff
@@ -155,3 +169,22 @@ func (a *AdminAPI) ToggleTopicPartitions(ctx context.Context, disabled bool, nam
 	}{disabled}
 	return a.sendToLeader(ctx, http.MethodPost, disableURL, body, nil)
 }
+
+// MajorityLost returns a list of partitions with lost majority given an input
+// set of dead nodes.
+func (a *AdminAPI) MajorityLost(ctx context.Context, deadNodes []int) ([]MajorityLost, error) {
+	var rr []MajorityLost
+	// One liner to convert []int -> int,int,int.
+	csv := strings.Trim(strings.Join(strings.Split(fmt.Sprint(deadNodes), " "), ","), "[]")
+	path := fmt.Sprintf("/v1/partitions/majority_lost?dead_nodes=%v", csv)
+	return rr, a.sendAny(ctx, http.MethodGet, path, nil, &rr)
+}
+
+// ForceRecoverFromNode Force recovers partitions from input list of nodes.
+func (a *AdminAPI) ForceRecoverFromNode(ctx context.Context, plan []MajorityLost, deadNodes []int) error {
+	body := map[string]interface{}{
+		"dead_nodes":                  deadNodes,
+		"partitions_to_force_recover": plan,
+	}
```

Comment:
I think both Rogger and I were expecting this to basically be:
* Get a list of partitions that lost majority.
* ask for those partitions to be recovered
* optionally: specify which node to recovery from, i.e. which one do we think has the most data

Why are including the following information when asking to recover:
* term
* replicas
* dead_nodes
* _as well as_ a top level dead_nodes
What does this mean?

cc @bharathv @mattschumpert @piyushredpanda

Reply:
Basically I would expect us _not_ to include per-partition term,replicas,dead_nodes in the POST here, and instead just include: for each partition, recover from this potential set of nodes.

Reply:
The base case here is explicitly _focusing_ on 1 or more nodes you _know_ are dead (not the entire cluster, though of course that could happen if the balancer moves around stuck partition replicas).  The main usage scenario is ' of 1...6, nodes 3 an 5 are dead, check them for unhealthy partitions to repair, give me a proposal listing the partitions and let me confirm the destructive operation before committing, in which I need to tell the operation which nodes to execute this on.

It's importantly a two-step operation as part of the requirements.

Reply:
> optionally: specify which node to recovery from, i.e. which one do we think has the most data

this is automatically done, it picks a replica with the most latest data and hydrates the remaining from that state.

> Why are including the following information when asking to recover:

We include the following information because the state of the topic/partitions can change from the time user 'GET's the information about partitions with lost majority, digests it and the submits the request to recover them..

consider the following sequence of actions.

example 1:

> (user 1) GET  partitions_with_lost_majority 
>  (user 2) delete foo
>  (user 2) create foo
>  (user 1) POST unsafe-recover --nodes 1

Now unsafe recover command force recovers an incorrect revision of foo (which is not what the user approved)

example 2.

> (user 1)  GET  partitions_with_lost_majority 
> (user 2) manual force recover foo
> (user 1)  POST unsafe-recover --nodes 1

Now unsafe recovery will force recover `foo` again even though it was already force recovered by user 2 manually

We include all the metadata information in the payload that ties the partitions to the same instances that the user submitted the approval for.


> topic_revision (not term)

This avoids example 1 above, because there is additional validation at the time of force recovery that the revision of the topic is same (it skips if the revision has advanced).

> replicas

This avoids example 2 and similar cases. If the replica set has changed from the time the user has approved, force recovery is skipped.

> as well as a top level dead_nodes

This is the same list of nodes that are included in the rpk command, this is persisted in the controller log for the record.


> dead_nodes

This is a subset of top level dead_nodes that are actually causing lost majority for a given partition. 

example:

partition replicas = [1]
dead_nodes = [1, 2]

top level dead nodes = [1, 2]
partition dead nodes = [1]